Label API 완성

moaikang opened the pull request:
### Issue Number
Close #84 
Close #81 
Close #79 
Close #77 

### 새로운 기능
- label CRUD

### 작업 유형
- [x] 신규 기능 추가
- [ ] 버그 수정
- [ ] 리펙토링
- [ ] 문서 업데이트

### 체크리스트
- [x] Merge 하는 브랜치가 올바른가?
- [x] 코딩컨벤션을 준수하는가?
- [x] PR과 관련없는 변경사항이 없는가?
- [x] 내 코드에 대한 자기 검토가 되었는가?
- [ ] 변경사항이 효과적이거나 동작이 작동한다는 것을 보증하는 테스트를 추가하였는가?
- [ ] 새로운 테스트와 기존의 테스트가 변경사항에 대해 만족하는가?

Commit:
feat    : :zap:️ API : GET /api/label

라벨을 전부 가져오는 API를 생성하였다.

Commit:
feat    : API Delete /api/label/:id

해당하는 id의 라벨을 삭제하는 API를 만들었다.

Commit:
feat    : :zap:️ API : PATCH /api/label/:id

해당하는 id와 일치하는 라벨의 정보를 수정하는 API를 만들었다.

namda-on commented on web/server/src/services/label.service.ts:
```diff
@@ -0,0 +1,51 @@
+
+
+
+  update: async (id: number, body: any): Promise<void> => {
+    console.log(body);
+    const labelRepository = getRepository(LabelEntity);
+    const label = await LabelService.getLabel(id);
+
+    if (!label) throw new Error("label does not exist");
+
+    await labelRepository.update(label, body);
```
body 도 class 나 인터페이스로 뭐가들어올지를 알려주면 더 좋지않을까요?ㅎㅎ

namda-on commented on web/server/src/routers/label.ts:
```diff
@@ -12,8 +12,65 @@ LabelRouter.post("/", async (req: Request, res: Response) => {
     return res.status(200).json(result);
   } catch (e) {
-    return res.status(400).json(e);
+    return res.status(400).json({
+      status: 400,
+      message: e.message
+    });
```
확실히 response 객체 자체에 status가 들어있는데 body 한번 더 보내는 것은 불필요해보이기는 하네요! 회의로 response 객체 형식에 대한 통일이 필요해봅니다!

namda-on reviewed (approved):
너무 고생하셨습니다!!  완벽하네요ㅎㅎ 🎃

Commit:
refactor: :rainbow: API label refactor

회의 때 정한 response 규칙으로 response를 리팩토링했다.
그리고, results로 되어있는 이름들을 좀 더 명확하게 리팩토링 했다.

namda-on commented on web/server/src/services/label.service.ts:
```diff
@@ -26,17 +26,13 @@ const LabelService = {
-
-
   update: async (id: number, body: any): Promise<void> => {
     console.log(body);
```
콘솔로그 발견했습니돵!!

Commit:
refactor: :rainbow: Label API response 형태 리팩토링

response형태를 새롭게 회의해서 label API를 리팩토링 하였다.

maong0927 commented on web/server/src/routers/label.ts:
```diff
@@ -1,18 +1,62 @@
+
+  try {
+    await LabelService.update(parseInt(id), body);
+    return res.status(200).json();
```
여기도용!!

maong0927 commented on web/server/src/routers/label.ts:
```diff
@@ -1,18 +1,62 @@
+
+  try {
+    const labels = await LabelService.getLabels();
+    return res.status(200).json(labels);
```
status(200)을 생략해 주시면 좋을 거 같습니다!

Commit:
refactor: :rainbow: 코드리뷰 리팩토링

팀원 분들이 해주신 코드리뷰를 기반으로 리팩토링 하였다.
